tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java

```diff
@@ -387,12 +390,51 @@ private Map<String, FlowNodeTree> buildFlowNodeTrees(final List<GetVariablesRequ
 
     final Map<String, FlowNodeTree> flowNodeTrees = new HashMap<>();
     for (final FlowNodeInstanceEntity flowNodeInstance : flowNodeInstances) {
-      getFlowNodeTree(flowNodeTrees, flowNodeInstance.getProcessInstanceId())
-          .setParent(flowNodeInstance.getId(), flowNodeInstance.getParentFlowNodeId());
+      getFlowNodeTree(flowNodeTrees, String.valueOf(flowNodeInstance.getProcessInstanceKey()))
+          .setParent(flowNodeInstance.getId(), String.valueOf(getScopeKey(flowNodeInstance)));
     }
+
+    // ensure that process instances are added
+    // as a FNI to the FNI tree
+    flowNodeTrees
+        .keySet()
+        .forEach(
+            id -> {
+              getFlowNodeTree(flowNodeTrees, id).setParent(id, ABSENT_PARENT_ID);
+            });
+
     return flowNodeTrees;
   }
 
+  private Long getScopeKey(final FlowNodeInstanceEntity flowNodeInstance) {
```

Comment:
Changes on this file seem to be same as in the [TaskVariableSearchUtil](https://github.com/camunda/camunda/pull/25386/files#diff-4691dc7faa3d9fe82bf63efd305f00fad95a35f10077a5156cb34407754b9ea7), maybe we can reuse them?

Reply:
Yeah, I had the same thinking, but I didn't want to put too much effort into it. I just want to make it work and avoid refactorings as much as possible.